draft-aegis-mimi-arch-latest.md

```diff
@@ -43,28 +43,166 @@ TODO Abstract
 
 # Introduction
 
-## Scope:
-* interface between AS and MLS -- something AS has to implement for both
-  client and back-end
-* interface between AS and application, split into 3 components
-* Overview of services and very high-level of how they are used together.
-* ?? how components federate
+In order to achieve cross-application federation between a set of MLS applications there must be a common set of APIs available that implement the basic AS/DS requirements as defined by `MLS Architecture`. To facilitate the creation of groups asynchronously, MLS clients may have a need to establish identity, credentials and a queue of key packages. Once groups are established, it is useful for clients to have access to a common repository for welcome messages, pending proposals, commit messages, and ratchet trees. 
 
-## Actors
-* back-end
-* application
-* client
+In this document we describe a MIMI Gateway API that can be used to provide intra domain and cross domain federation between MLS applications.
 
-// make clear that client never talks to backend and that app should
-// implement access control itself
+# Operational Context 
 
-# AS - MLS Interface
+A basic federation scenario consists of a bi-directional data flow between actors. Clients
+communicate with application servers, application servers
+communicate with gateway services, and gateway services
+communicate with other gateway services.
+
+~~~
++ - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - +
+'                       Domain A                                '
+'                                                               '
+' +-------+     +----------------------+     +----------------+ '
+' | Alice | --- | Application Server 1 | --- | MIMI Gateway A |-'-----+
+' +-------+     +----------------------+     +----------------+ '     |

```

Comment:
Would it help to show that a gateway is useful even if an app doesn't federate with anything?

Reply:
I think separate diagram, but yeah that makes sense

Reply:
Created #1